### PR TITLE
Shifted methods to the correct subpackage

### LINKED ISSUE #2373: Some methods are in the wrong sub-package
1. `SHLDDRK_2N` and `SHLDDRK52` should be moved from OrdinaryDiffEqSSPRK.jl to OrdinaryDiffEqLowStorageRK.jl since they are not SSP
2. `KYK2014DGSSPRK_3S2` should be moved from OrdinaryDiffEqLowStorageRK.jl to OrdinaryDiffEqSSPRK.jl since the other SSP methods are there, even if they are low storage (which most of them are)
3. The docstring of `ssp_coefficient` should move from OrdinaryDiffEqLowStorageRK.jl to OrdinaryDiffEqSSPRK.jl or OrdinaryDiffEqCore.jl

### Details

https://github.com/SciML/OrdinaryDiffEq.jl/blob/dc0e1e733fdd79a6c643cbd0c4f4be03c934cb91/lib/OrdinaryDiffEqSSPRK/src/OrdinaryDiffEqSSPRK.jl#L102

`SHLDDRK_2N` and `SHLDDRK52` are not SSP methods but low-storage methods. There is also no `ssp_coefficient` defined for them, see 

https://github.com/SciML/OrdinaryDiffEq.jl/blob/dc0e1e733fdd79a6c643cbd0c4f4be03c934cb91/lib/OrdinaryDiffEqSSPRK/src/alg_utils.jl#L36-L52

https://github.com/SciML/OrdinaryDiffEq.jl/blob/dc0e1e733fdd79a6c643cbd0c4f4be03c934cb91/lib/OrdinaryDiffEqLowStorageRK/src/OrdinaryDiffEqLowStorageRK.jl#L107

`KYK2014DGSSPRK_3S2` is an SSP method, see also

https://github.com/SciML/OrdinaryDiffEq.jl/blob/dc0e1e733fdd79a6c643cbd0c4f4be03c934cb91/lib/OrdinaryDiffEqLowStorageRK/src/alg_utils.jl#L86-L100

### PR DESCRIPTION
**Pull request by ParamThakkar123**
## Checklist

- [X] Appropriate tests were added
- [X] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [X] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

Fixes #2373

### CODE REVIEW

File: lib/OrdinaryDiffEqRosenbrock/src/interp_func.jl
Comment (ranocha) [inline lib/OrdinaryDiffEqRosenbrock/src/interp_func.jl:1]: There are a lot of changes to the Rosenbrock files - they are not intended here, are they?

Comment (ParamThakkar123) [reply lib/OrdinaryDiffEqRosenbrock/src/interp_func.jl:1]: The commits made to the rosenbrock files are part of #2320. They came here when I switched branches

Comment (ChrisRackauckas) [reply lib/OrdinaryDiffEqRosenbrock/src/interp_func.jl:1]: Please branch off of master. These changes should not be mixed.

Comment (ranocha) [reply lib/OrdinaryDiffEqRosenbrock/src/interp_func.jl:1]: You may use `git cherry-pick` when doing this

Comment (ParamThakkar123) [reply lib/OrdinaryDiffEqRosenbrock/src/interp_func.jl:1]: I started a new PR #2408 after cleaning the master and branching it off

**Pull request closed without merging**